src/server/server.rs

```diff
@@ -77,6 +78,10 @@ pub struct Server<T: RaftStoreRouter + 'static, S: StoreAddrResolver> {
     end_point: EndPointHost,
 
     resolver: S,
+
+    // TODO: remove `#[allow(dead_code)]`
+    #[allow(dead_code)]
+    metric: Metric,
```

Comment:
I think you should use Arc<Metric>, we may use it in other places, like rafestore.